File: src/codewars/python/7kyu/find_sum.py
"""

Your task is to write function findSum.

Upto and including n, this function will return the sum of all multiples of 3 and 5.

For example:

findSum(5) should return 8 (3 + 5)

findSum(10) should return 33 (3 + 5 + 6 + 9 + 10)


"""

import unittest

def find(n):
    three_multi_list = multiply(3, n)
    five_multi_list = multiply(5, n)
    return sum(three_multi_list) + sum(five_multi_list)


def multiply(m, n):
    counter = 1
    num_list = []
    multi = m * counter
    while multi <= n:
        num_list.append(multi)
        counter += 1
        multi = m * counter

    return num_list


class TestCase(unittest.TestCase):

    def test_find(self):
        self.assertEqual(find(10), 33)
        self.assertEqual(find(100), 2418)



if __name__ == "__main__":
    unittest.main()

File: src/codewars/ruby/7kyu/find_sum.rb
"""

Your task is to write function findSum.

Upto and including n, this function will return the sum of all multiples of 3 and 5.

For example:

findSum(5) should return 8 (3 + 5)

findSum(10) should return 33 (3 + 5 + 6 + 9 + 10)



"""


def find(n)
  num_array = []
  counter = 1
  m_of_three = 1
  m_of_five = 1
  while m_of_three < n
    m_of_three = 3 * counter
    num_array << m_of_three if m_of_three <= n
    counter += 1
  end
  counter = 1
  while m_of_five < n
    m_of_five = 5 * counter
    num_array << m_of_five if m_of_five <= n
    counter += 1
  end
  return num_array.inject { |sum, x| sum + x }
end


describe "find" do
  context "given 5" do
    it "returns 8" do
      expect(find(5)).to eq(8)
    end
  end

  context "given 10" do
    it "returns 33" do
      expect(find(10)).to eq(33)
    end
  end

end
